core/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java

```diff
@@ -24,8 +27,81 @@
  */
 @RunWith(JUnit4.class)
 public class StatsContextFactoryTest {
-  @Test(expected = IOException.class)
+  private static final int VALUE_TYPE_STRING = 0;
+  private static final int VALUE_TYPE_INTEGER = 1;
+  private static final int VALUE_TYPE_BOOLEAN = 2;
+
+  private static final String KEY1 = "Key";
+  private static final String VALUE_STRING = "String";
+  private static final int VALUE_INT = 10;
+  private static final boolean VALUE_BOOL = true;
+
+  @Test
   public void testDeserializeEmpty() throws Exception {
-    Stats.getStatsContextFactory().deserialize(new ByteArrayInputStream(new byte[0]));
+    testDeserialize(new ByteArrayInputStream(new byte[0]));
```

Comment:
Since this is a test and doesn't throw an exception, you should test that the value is the expected value. If you think this is redundant with other tests, you can consider removing it.